src/main/java/com/github/manolo8/darkbot/core/manager/HeroManager.java

```diff
@@ -139,7 +138,7 @@ public void update(long address) {
 
         pet.update(API.readMemoryLong(address + 176));
         clickable.setRadius(0);
-        id = API.readMemoryInt(address + 56);
+        id = API.readMemoryInt(address + 56); // why id is read here?
```

Comment:
my guess for this is that it maybe wasn't updated for other entities or the location is different, but this code is pretty ancient anyways

check git blame for when it was last edited and figure out what the code was at that point in time
if this id is updated for all entities, you can try to remove it and see if it still works as intended

lastly, it could be that other entities' id never changes but hero entity maybe perssists, but its id changes somehow?